### PR TITLE
[13.0][MIG] project_task_dependency: Migration to 13.0 

### PR DESCRIPTION
**Pull request by manuelcalerosolis**
This is the migration to v 13.0 module mig_project_task_dependency

@Tecnativa TT19796

### CODE REVIEW

File: project_task_dependency/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# © 2016 Onestein (<http://www.onestein.eu>)
```
Comment (Tardo) [inline project_task_dependency/__manifest__.py:1]:
```suggestion
# Copyright 2016 Onestein (<http://www.onestein.eu>)
```

**Review by Tardo — CHANGES_REQUESTED** [review]
:+1: LGTM Good work! Only a little comment...

File: project_task_dependency/tests/test_project_task_dependency.py [hunk 2]
```diff
@@ -71,11 +71,10 @@ def test_copy(self):
...
-        old_count = self.env["project.task.copy.map"].search_count(
+        self.env["project.task.copy.map"].search_count(
             [("old_task_id", "=", self.task4.id)]
         )
         self.task4.copy()
-        new_count = self.env["project.task.copy.map"].search_count(
+        self.env["project.task.copy.map"].search_count(
             [("old_task_id", "=", self.task4.id)]
         )
-        self.assertEqual(old_count, new_count)
```
Comment (pedrobaeza) [inline project_task_dependency/tests/test_project_task_dependency.py]: Why are you removing this assert? This gets orphaned with it

File: project_task_dependency/__manifest__.py (hunk 1 above)
Comment (manuelcalerosolis) [reply project_task_dependency/__manifest__.py:1]: Thanks!

File: project_task_dependency/tests/test_project_task_dependency.py (hunk 2 above)
Comment (manuelcalerosolis) [reply project_task_dependency/tests/test_project_task_dependency.py]: My mistake, fixed!

File: project_task_dependency/README.rst [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+* Onestein
+
+Contributors
+~~~~~~~~~~~~
+
+* Dennis Sluijk <d.sluijk@onestein.nl>
+* Adria Gil Sorribes <adria.gil@eficent.com>
+* Pedro M. Baeza <pedro.baeza@tecnativa.com>
+* Tecnativa - Manuel Calero
```
Comment (ernestotejeda) [inline project_task_dependency/README.rst]:
```suggestion
* `Tecnativa <https://www.tecnativa.com>` _:

  * Manuel Calero
```
We usually write that this way:

**Review by ernestotejeda — APPROVED** [review]
Code OK
Functionally OK

Just a suggestion related to Contributor section in README.rst

File: project_task_dependency/models/project_task.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    @api.constrains("dependency_task_ids")
+    def _check_dependency_recursion(self):
+        if not self._check_m2m_recursion("dependency_task_ids"):
+            raise ValidationError(
+                _("You cannot create recursive dependencies between tasks.")
+            )
+
+    def copy(self, default=None):
+        self.ensure_one()
+        res = super(ProjectTask, self).copy(default)
```
Comment (ernestotejeda) [inline project_task_dependency/models/project_task.py:80]:
```suggestion
        res = super().copy(default)
```

**Review by ernestotejeda — CHANGES_REQUESTED** [review]

**Review by emagdalenaC2i — APPROVED** [review]
LGTM only attend pending comments

**Review by astirpe — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@manuelcalerosolis please perform the suggested changes and I'll merge.

File: project_task_dependency/README.rst (hunk 3 above)
Comment (manuelcalerosolis) [reply project_task_dependency/README.rst]: Done

File: project_task_dependency/models/project_task.py (hunk 4 above)
Comment (manuelcalerosolis) [reply project_task_dependency/models/project_task.py:80]: Done

**Pull request merged**